solutions/gold/usaco-993.mdx

```diff
@@ -2,14 +2,28 @@
 id: usaco-993
 source: USACO Gold 2020 January
 title: Time is Mooney
-author: Nathan Gong, Ryan Chou
+author: Nathan Gong, Ryan Chou, David Guo
 ---
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_time_gold_jan20.html)
 
 ## Explanation
 
-We only need to keep track of the number of days that Bessie has been traveling and the amount of mooney that she has made. For every day up to $T_{\text{max}}$, we'll try to travel to each city, and update the mooney made if the current path is more optimal. The maximum amount of money she makes is $1000t - t^2$, in the case where she makes $1000$ mooney per city and the cost is $1$. Her earnings become negative when $t > 1000$, so we only have to check her movement across the cities for at most $1000$ days.
+We define $dp[t][i]$ as the maximum moonies Bessie can have at city $i$ on day $t$.
+Starting with $dp[0][0] = 0$, we iterate over each day up to $T_{\text{max}}$,
+the maximum number of days, and over each city, updating $dp[t + 1][j]$
+for all cities $j$ reachable from city $i$ via a directed road.
+
+We compare the current value of $dp[t + 1][j]$ with the
+value of coming from an adjacent city on the previous day,
+$dp[t][i] + \text{moonies}[j]$, and take the maximum.
+After processing each day, we calculate the profit as $dp[t][0] - C \cdot t^2$,
+representing the total moonies minus the travel cost for $t$ days,
+and track the maximum profit.
+
+Note that the maximum amount of money Bessie makes is $1000t - t^2$,
+in the case where she makes $1000$ mooney per city and the cost is $1$ mooney.
+Her earnings become negative when $t > 1000$, so we only have to check her movement across the cities for at most $1000$ days.
```

Comment:
```suggestion
We define $\texttt{dp}[t][i]$ as the maximum moonies Bessie can have at city $i$ on day $t$.
Starting with $\texttt{dp}[0][0] = 0$, we iterate over each day up to $T_{\text{max}}$,
the maximum number of days, and over each city, updating $\texttt{dp}[t + 1][j]$
for all cities $j$ reachable from city $i$ via a directed road.

We compare the current value of $\texttt{dp}[t + 1][j]$ with the
value of coming from an adjacent city on the previous day,
$\texttt{dp}[t][i] + \text{moonies}[j]$, and take the maximum.
After processing each day, we calculate the profit as $\texttt{dp}[t][0] - C \cdot t^2$,
representing the total moonies minus the travel cost for $t$ days,
and track the maximum profit.

Note that the maximum amount of money Bessie makes is $1000t - t^2$,
in the case where she makes $1000$ mooney per city and the cost is $1$ mooney.
Her earnings become negative when $t > 1000$, so we only have to check her movement across the cities for at most $1000$ days.
```

Reply:
Also mention that the maximum amount is $1000t-t^2$ because the edge weights $m_i$ are bounded by $1000$?